src/myrtlespeech/model/hard_lstm.py

```diff
@@ -0,0 +1,419 @@
+        forget_gate_bias=forget_gate_bias,
+        batch_first=batch_first,
+        bidirectional=bidirectional,
+        bias=bias,
+    )
+
+
+class StackedLSTM(torch.nn.Module):
+    """Multi-layer LSTM.
+
+    Args:
+        num_layers: See :py:class:`HardLSTM`.
+
+        layer_type: The layer type used in the :py:class:`StackedLSTM`.
+
+        input_size: See :py:class:`HardLSTM`.
+
+        hidden_size: See :py:class:`HardLSTM`.
+
+        bias: See :py:class:`HardLSTM`.
+
+        dropout: See :py:class:`HardLSTM`.
+
+        bidirectional: See :py:class:`HardLSTM`.
+
+        forget_gate_bias: See :py:class:`HardLSTM`.
+
+        batch_first: See :py:class:`HardLSTM`.
+    """
+
+    def __init__(
+        self,
+        num_layers: int,
+        layer_type: torch.nn.Module,
+        input_size: int,
+        hidden_size: int,
+        bias: int = True,
+        dropout: float = 0.0,
+        bidirectional: bool = False,
+        forget_gate_bias: Optional[float] = None,
+        batch_first: bool = False,
+    ):
+
+        super(StackedLSTM, self).__init__()
+        self.input_size = input_size
+        self.hidden_size = hidden_size
+        self.num_layers = num_layers
+        self.num_directions = num_directions = 2 if bidirectional else 1
+        self.batch_first = batch_first
+        self.bias = bias
+        self.dropout = dropout
+        self.bidirectional = bidirectional
+
+        first_layer_args = [
+            input_size,
+            hidden_size,
+            forget_gate_bias,
+        ]
+        other_layer_args = [
+            hidden_size * num_directions,
+            hidden_size,
+            forget_gate_bias,
+        ]
+
+        layers = [layer_type(*first_layer_args)] + [
+            layer_type(*other_layer_args) for _ in range(num_layers - 1)
+        ]
+
+        self.layers = torch.nn.ModuleList(layers)
+
+    def forward(
+        self, input: Tensor, hx: Tuple[Tensor, Tensor]
+    ) -> Tuple[Tensor, Tuple[Tensor, Tensor]]:
+        """Performs forward pass of LSTM.
+
+        All inputs are moved to the GPU with :py:meth:`torch.nn.Module.cuda`
+        if :py:func:`torch.cuda.is_available` was :py:data:`True` on
+        initialisation.
+
+        Args:
+            input: A :py:class:`torch.Tensor` of rnn sequence inputs.
+
+            hx: The lstm hidden state: a Tuple of hidden state and cell state.
+
+        Returns:
+            A Tuple of Tuples; ``res = a, (b, c)`` where ``a`` is the lstm
+                output and ``(c, d)`` are the hidden and cell states of the
+                lstm.
+        """
+
+        if self.batch_first:
+            input = input.transpose(0, 1)
+
+        hn, cn = hx
+        req_size = self.num_layers, self.num_directions, hn.size(1), hn.size(2)
+        hn = hn.view(req_size)
+        cn = hn.view(req_size)
+
+        i = 0
+        output = input
+        hn_out = torch.jit.annotate(List[Tensor], [])
+        cn_out = torch.jit.annotate(List[Tensor], [])
```

Comment:
See comment about switching this to use PyTorch-style type annotations rather than `torch.jit.annotate`.